docs: refactored the dropdownable component story issue number #1004

electron97 opened the pull request:
<!-- PR Checklist -->

# Description
 - Currently, The `Dropdownable` component's story is configured so that it is wrapped in the `<DropdownStory />`. This is used to manage the state and functionality of the component.
 - The Code option in the component Docs page serves as a reference for how to use the component. With this HOC component wrapping it we cannot see the example of how to use this component.
 - To fix this we need to move `DropdownStory` logic into `Dropdownable.Stories.tsx` so that it allows for the Dropdownable component code to fully display.
 - We also saw a similar issue with `DropdownStoryFit` as well so we moved it to `Dropdownable.Stories.tsx`.

## Which issue(s) does this PR relate to?

https://github.com/d2iq/ui-kit/issues/1004

## Testing

## Screenshots

|View| Before | After|
|---|---|---|
|DropdownStory|<img width="1470" alt="Screenshot 2023-10-05 at 4 21 33 PM" src="https://github.com/d2iq/ui-kit/assets/132813311/38268dd8-4bab-43c2-a472-5e4fdd12672c">|<img width="1468" alt="Screenshot 2023-10-05 at 4 16 29 PM" src="https://github.com/d2iq/ui-kit/assets/132813311/816d933f-b973-4b68-93cc-fb5f3608a43a">|
|DropdownStoryFit|<img width="1470" alt="Screenshot 2023-10-05 at 4 21 53 PM" src="https://github.com/d2iq/ui-kit/assets/132813311/5ab7ed62-e879-4379-83f6-7ab267407597">|<img width="1470" alt="Screenshot 2023-10-05 at 4 16 50 PM" src="https://github.com/d2iq/ui-kit/assets/132813311/2be0745f-0e21-4fda-af4c-b1fc3981eb28">|


## Checklist


- [ ] This PR is associated with a JIRA and it is mentioned in the commit message footer ("Closes …")
- [ ] Significant changes have been tested downstream to avoid breaking changes
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [x] I have reviewed the changes and provided detail to the sections above

nataliepina commented:
Thank you for including the extra clean up work here! 

At first glance the code all looks good. However, I am attempting to test this out in Storybook and encountering some issues. It seems the story is completely frozen now and I am unable to see the component. This is making it hard to debug, but I am looking into it. Have you encountered this?

This is what I am seeing:
<img width="1463" alt="Screenshot 2023-10-05 at 8 49 54 AM" src="https://github.com/d2iq/ui-kit/assets/34781875/a2ae0900-2cc7-4fb3-b30d-f3fcd2680dbf">

I am seeing it when running locally and with the Uffizzi link. 
Are you able to encounter this same issue with this link -- https://pr-1047-deployment-37707-ui-kit.app.uffizzi.com/?path=/docs/utils-dropdownable--docs

electron97 commented:
Hey @nataliepina  ✋ ,

Apologies for the confusion. I had added an `export` to the `const` variable by mistake which was causing this issue. I have added a commit to fix this. This should work fine now. Please let me know if the issue still isn't resolved.

Thanks!

nataliepina reviewed (approved):
It's fixed for me now. Great work, thank you for contributing. Happy Hacktoberfest!
